src/main/java/com/devonfw/quarkus/productmanagement/service/v1/ProductRestService.java

```diff
@@ -20,92 +20,54 @@
-//default is Singleton scope
-@Path("/products")
-// how we serialize response
+@Path("/product/v1")
 @Produces(MediaType.APPLICATION_JSON)
-// how we deserialize params
 @Consumes(MediaType.APPLICATION_JSON)
 public class ProductRestService {
 
-  // using @Context we can inject contextual info from JAXRS(e.g. http request, current uri info, endpoint info...)
-  @Context
-  UriInfo uriInfo;
-
   @Inject
   UcFindProduct ucFindProduct;
 
   @Inject
-  UcManageProduct ucManageProduct;
-
-  @APIResponses({
-  @APIResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = PagedProductResponse.class))),
-  @APIResponse(responseCode = "500") })
-  @Operation(operationId = "Get Products", description = "Returns list of Products matching given criteria, uses pagination")
-  @GET
-  // REST service methods should not declare exceptions, any thrown error will be transformed by exceptionMapper in
-  // tkit-rest
-  // We did not define custom @Path - so it will use class level path
-  public Page<ProductDto> getAll(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProducts(dto);
-  }
-
-  @GET
-  @Path("criteriaApi")
-  public Page<ProductDto> getAllCriteriaApi(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProductsByCriteriaApi(dto);
-  }
-
-  @GET
-  @Path("queryDsl")
-  public Page<ProductDto> getAllQueryDsl(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProductsByQueryDsl(dto);
-  }
+  UcDelete ucDelete;
 
-  @GET
-  @Path("query")
-  public Page<ProductDto> getAllQuery(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProductsByTitleQuery(dto);
-  }
+  @Inject
+  UcSave ucSave;
 
-  @GET
-  @Path("nativeQuery")
-  public Page<ProductDto> getAllNativeQuery(@BeanParam ProductSearchCriteriaDto dto) {
+  @POST
+  @Path("search")
+  public Page<ProductDto> getAllQueryDsl(@BeanParam ProductSearchCriteriaDto productSearch) {
 
-    return this.ucFindProduct.findProductsByTitleNativeQuery(dto);
+    return this.ucFindProduct.findProducts(productSearch);
   }
 
   @GET
-  @Path("ordered")
   public Page<ProductDto> getAllOrderedByTitle() {
 
-    return this.ucFindProduct.findProductsOrderedByTitle();
+    return this.ucFindProduct.findProductsOrderByTitle();
   }
 
   @APIResponses({
-  @APIResponse(responseCode = "200", description = "OK, New Product created", content = @Content(schema = @Schema(implementation = NewProductDto.class))),
+  @APIResponse(responseCode = "201", description = "OK, New Product created", content = @Content(schema = @Schema(implementation = ProductDto.class))),
   @APIResponse(responseCode = "400", description = "Client side error, invalid request"),
   @APIResponse(responseCode = "500") })
-  @Operation(operationId = "createNewProduct", description = "Stores new Product in DB")
+  @Operation(operationId = "save", description = "Stores new Product in DB")
```

Comment:
Restful guidelines: `POST /resource` = create new resources in the collection of resources. It is not a save operation. Also it is not idempotent. Hence the previous `createNewProduct` was more correct

Reply:
All these OpenAPI annotations lead to pointless maintenance overhead and are fully redundant to what is already in the code. AFAIK we decided to generate this instead of adding all these annotations.